programs/stake/src/stake_state.rs

```diff
@@ -1486,7 +1496,7 @@ impl MergeKind {
                     source_lamports,
                     source_stake.credits_observed,
                 )?;
-                Some(StakeState::Stake(meta, stake))
+                Some(StakeState::Stake(meta, stake, DeactivationFlag::Empty))
```

Comment:
Same issue here -- you'll have to propagate the "highest" deactivation flag, ie. if you merge `Empty` into `MustFullyActivate...` or vice versa, the resulting stake must have `MustFullyActivate...`